core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -502,6 +503,15 @@ private void registerBuiltinSorts() {
         namedWriteableRegistry.register(SortBuilder.class, FieldSortBuilder.NAME, FieldSortBuilder::new);
     }
 
+    private void registerBuiltinValueFormats() {
+        namedWriteableRegistry.register(ValueFormat.class, ValueFormat.Boolean.NAME, ValueFormat.Boolean::new);
+        namedWriteableRegistry.register(ValueFormat.class, ValueFormat.DateTime.NAME, ValueFormat.DateTime::new);
+        namedWriteableRegistry.register(ValueFormat.class, ValueFormat.Decimal.NAME, ValueFormat.Decimal::new);
+        namedWriteableRegistry.register(ValueFormat.class, ValueFormat.GeoHash.NAME, ValueFormat.GeoHash::new);
+        namedWriteableRegistry.register(ValueFormat.class, ValueFormat.Ip.NAME, ValueFormat.Ip::new);
+        namedWriteableRegistry.register(ValueFormat.class, ValueFormat.Raw.NAME, ValueFormat.Raw::new);
```

Comment:
should we have a `registerValueFormat(String name, Reader reader)` method so plugins that add a new field type can register the value type? If we do add that method, the built in value formats should be registered using that method too.